compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -971,10 +987,54 @@ public void visit(BLangTernaryExpr ternaryExpr) {
         analyzeExpr(ternaryExpr.elseExpr);
     }
 
-    public void visit(BLangAwaitExpr awaitExpr) {
-        analyzeExpr(awaitExpr.expr);
+    public void visit(BLangWaitExpr awaitExpr) {
+        analyzeExpr(awaitExpr.getExpression());
```

Comment:
When visiting the binary expression for wait for any recursively find the parent and validate if it is from wait or not. Allow only if the parent is wait else throw an error?